File: php/tests/Fixtures.php
<?php


namespace TransactionAnalyser\Tests;


use Brick\DateTime\LocalDateTime;
use Brick\Math\BigDecimal;
use DateTime;
use TransactionAnalyser\Transaction;
use TransactionAnalyser\TransactionType;

class Fixtures
{
    static function transactionData(): array
    {
        return [
            new Transaction(
                "WLMFRDGD",
                LocalDateTime::fromDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 12:45:33")),
                BigDecimal::of(59.99),
                "Kwik-E-Mart",
                new TransactionType("PAYMENT"),
                ""
            ),
            new Transaction(
                "YGXKOEIA",
                LocalDateTime::fromDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 12:46:17")),
                BigDecimal::of(10.95),
                "Kwik-E-Mart",
                new TransactionType("PAYMENT"),
                ""
            ),
            new Transaction(
                "LFVCTEYM",
                LocalDateTime::fromDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 12:50:02")),
                BigDecimal::of(5.00),
                "MacLaren",
                new TransactionType("PAYMENT"),
                ""
            ),
            new Transaction(
                "SUOVOISP",
                LocalDateTime::fromDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 13:12:22")),
                BigDecimal::of(5.00),
                "Kwik-E-Mart",
                new TransactionType("PAYMENT"),
                ""
            ),
            new Transaction(
                "AKNBVHMN",
                LocalDateTime::fromDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 13:14:11")),
                BigDecimal::of(10.95),
                "Kwik-E-Mart",
                new TransactionType("REVERSAL"),
                "YGXKOEIA"
            ),
            new Transaction(
                "JYAPKZFZ",
                LocalDateTime::fromDateTime(DateTime::createFromFormat('d/m/Y H:i:s', "20/08/2020 14:07:10")),
                BigDecimal::of(99.50),
                "MacLaren",
                new TransactionType("PAYMENT"),
                ""
            ),
        ];
    }
}

File: transaction-analyzer/csharp/TransactionAnalyzer.Lib.Tests/Fixtures.cs
using System;
using System.Globalization;

namespace TransactionAnalyzer.Lib.Tests;

internal class Fixtures
{
    public static Transaction[] Load()
    {
        return new Transaction[]
        {
            new()
            {
                Id = "WLMFRDGD",
                TransactedAt = DateTime.ParseExact("20/08/2020 12:45:33", "dd/MM/yyyy HH:mm:ss",
                    CultureInfo.InvariantCulture,
                    DateTimeStyles.None),
                Amount = new decimal(59.99),
                MerchantName = "Kwik-E-Mart",
                Type = TransactionType.PAYMENT,
                RelatedTransactionId = ""
            },
            new()
            {
                Id = "YGXKOEIA",
                TransactedAt = DateTime.ParseExact("20/08/2020 12:46:17", "dd/MM/yyyy HH:mm:ss",
                    CultureInfo.InvariantCulture,
                    DateTimeStyles.None),
                Amount = new decimal(10.95),
                MerchantName = "Kwik-E-Mart",
                Type = TransactionType.PAYMENT,
                RelatedTransactionId = ""
            },

            new()
            {
                Id = "LFVCTEYM",
                TransactedAt = DateTime.ParseExact("20/08/2020 12:50:02", "dd/MM/yyyy HH:mm:ss",
                    CultureInfo.InvariantCulture,
                    DateTimeStyles.None),
                Amount = new decimal(5.00),
                MerchantName = "MacLaren",
                Type = TransactionType.PAYMENT,
                RelatedTransactionId = ""
            },

            new()
            {
                Id = "SUOVOISP",
                TransactedAt = DateTime.ParseExact("20/08/2020 13:12:22", "dd/MM/yyyy HH:mm:ss",
                    CultureInfo.InvariantCulture,
                    DateTimeStyles.None),
                Amount = new decimal(5.00),
                MerchantName = "Kwik-E-Mart",
                Type = TransactionType.PAYMENT,
                RelatedTransactionId = ""
            },

            new()
            {
                Id = "AKNBVHMN",
                TransactedAt = DateTime.ParseExact("20/08/2020 13:14:11", "dd/MM/yyyy HH:mm:ss",
                    CultureInfo.InvariantCulture,
                    DateTimeStyles.None),
                Amount = new decimal(10.95),
                MerchantName = "Kwik-E-Mart",
                Type = TransactionType.REVERSAL,
                RelatedTransactionId = "YGXKOEIA"
            },

            new()
            {
                Id = "JYAPKZFZ",
                TransactedAt = DateTime.ParseExact("20/08/2020 14:07:10", "dd/MM/yyyy HH:mm:ss",
                    CultureInfo.InvariantCulture,
                    DateTimeStyles.None),
                Amount = new decimal(99.50),
                MerchantName = "MacLaren",
                Type = TransactionType.PAYMENT,
                RelatedTransactionId = ""
            }
        };
    }
}